test/classes/diten/instantiatedClassname.chpl

```diff
@@ -4,9 +4,10 @@ class C {
   param c: int = 3;
 }
 
-proc foo(c: borrowed C(?)?) {
+proc foo(c: borrowed C(?)) {
   writeln(c.b);
 }
 
-var c: borrowed C(b = 1)?;
+var own = new C(b = 1);
+var c: borrowed C(b = 1) = own;
```

Comment:
The test still works as it was originally written. I view the changes to this test as just a clean-up. I wrote this about my changes to this test:

    Adjust instantiatedClassname.chpl to avoid nilable class
    
    This test originated in 942157fc18e07f025891534b8eb8a4ed9a973917 as a bug
    report about class names for generic classes. This commit changes it to
    no longer use a nilable class that is default initialized (since that
    seems to be unrelated to the original purpose of the test).

I view the `?` in this test as an artifact of our migration process as the language design has changed over time. However I would be happy to add a sibling test that includes the `?` version if you feel that is important.

Reply:
I understand it better now, thank you.

I see that this test ensures that `c.b` works even when `c` is generic and nilable. Since we would like to preserve the latter as a convenience feature, let us keep a test for it. It could be a sibling test or additional code in the same test. Thanks!